apps/client/src/scenes/Benchmarks/Benchmarks.tsx

```diff
@@ -37,105 +38,203 @@ export default function Benchmarks() {
+        startTimer();
+        return api.differentArtistsPer(
+          interval.start,
+          interval.end,
+          Timesplit.all,
+        );
+      },
     },
     {
-      id: "getTimePerHourOfDay",
       title: "Get time per hour of day",
-      request: () => api.timePerHourOfDay(interval.start, interval.end),
+      request: startTimer => {
+        startTimer();
+        return api.timePerHourOfDay(interval.start, interval.end);
+      },
     },
     {
-      id: "getBestSongs",
       title: "Get best songs",
-      request: () => api.getBestSongs(interval.start, interval.end, 30, 1),
+      request: startTimer => {
+        startTimer();
+        return api.getBestSongs(interval.start, interval.end, NB, OFFSET);
+      },
     },
     {
-      id: "getBestArtists",
       title: "Get best artists",
-      request: () => api.getBestArtists(interval.start, interval.end, 30, 1),
+      request: startTimer => {
+        startTimer();
+        return api.getBestArtists(interval.start, interval.end, NB, OFFSET);
+      },
     },
     {
-      id: "getBestAlbums",
       title: "Get best albums",
-      request: () => api.getBestAlbums(interval.start, interval.end, 30, 1),
+      request: startTimer => {
+        startTimer();
+        return api.getBestAlbums(interval.start, interval.end, NB, OFFSET);
+      },
     },
     {
-      id: "getBestSongsOfHour",
       title: "Get best songs of hour",
-      request: () => api.getBestSongsOfHour(interval.start, interval.end),
+      request: startTimer => {
+        startTimer();
+        return api.getBestSongsOfHour(interval.start, interval.end);
+      },
     },
     {
-      id: "getBestAlbumsOfHour",
       title: "Get best albums of hour",
-      request: () => api.getBestAlbumsOfHour(interval.start, interval.end),
+      request: startTimer => {
+        startTimer();
+        return api.getBestAlbumsOfHour(interval.start, interval.end);
+      },
     },
     {
-      id: "getBestArtistsOfHour",
       title: "Get best artists of hour",
-      request: () => api.getBestArtistsOfHour(interval.start, interval.end),
+      request: startTimer => {
+        startTimer();
+        return api.getBestArtistsOfHour(interval.start, interval.end);
+      },
     },
     {
-      id: "getLongestSessions",
       title: "Get longest sessions",
-      request: () => api.getLongestSessions(interval.start, interval.end),
+      request: startTimer => {
+        startTimer();
+        return api.getLongestSessions(interval.start, interval.end);
+      },
+    },
+    {
+      title: "Get artist page",
+      request: async startTimer => {
+        const { data: bestArtists } = await api.getBestArtists(
+          interval.start,
+          interval.end,
+          1,
+          0,
+        );
+        const [bestArtist] = bestArtists;
+        if (!bestArtist) {
+          return;
+        }
+        startTimer();
+        return api.getArtistStats(bestArtist.artist.id);
+      },
+    },
```

Comment:
Instead of having the `startTimer` in all methods, maybe having an optional `preRequest` field that returns data reinjected in the `request` field? That would make back all other requests to a more compact form

Reply:
Okay I implemented what you talked about